src/query/sql/src/executor/physical_plans/physical_hash_join.rs

```diff
@@ -391,6 +391,7 @@ impl PhysicalPlanBuilder {
         let merged_fields = match join.join_type {
             JoinType::Cross
             | JoinType::Inner
+            | JoinType::AsOf
```

Comment:
Better to put it in a new branch: `JoinType::AsOf => unreachable!("Invalid join type {} during building physical hash join.", join.join_type)`.